lib/src/Uncertainty/Distribution/TruncatedDistribution.cxx

```diff
@@ -227,74 +204,82 @@ Point TruncatedDistribution::getRealization() const
   for (;;)
   {
     Point realization(distribution_.getRealization());
-    if ((!finiteLowerBound_ || realization[0] >= lowerBound_) && (!finiteUpperBound_ || realization[0] <= upperBound_)) return realization;
+    if (bounds_.contains(realization)) return realization;
   }
 }
 
 /* Get the DDF of the distribution: DDF_trunc = 1[a, b] * DDF / (CDF(b) - CDF(a)) */
 Point TruncatedDistribution::computeDDF(const Point & point) const
 {
-  if (point.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=1, here dimension=" << point.getDimension();
+  if (point.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=" << getDimension() << ", here dimension=" << point.getDimension();
 
-  const Scalar x = point[0];
-  if ((x <= lowerBound_) || (x > upperBound_)) return Point(1, 0.0);
+  if (!bounds_.contains(point)) return Point(getDimension(), 0.0);
   return normalizationFactor_ * distribution_.computeDDF(point);
 }
 
 
 /* Get the PDF of the distribution: PDF_trunc = 1[a, b] * PDF / (CDF(b) - CDF(a)) */
 Scalar TruncatedDistribution::computePDF(const Point & point) const
 {
-  if (point.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=1, here dimension=" << point.getDimension();
+  if (point.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=" << getDimension() << ", here dimension=" << point.getDimension();
+
+  Interval offset(Point(getDimension(), -quantileEpsilon_), Point(getDimension(), quantileEpsilon_));
+  if (!(bounds_ + offset).contains(point)) return 0.0;
 
-  const Scalar x = point[0];
-  if ((x < lowerBound_ - quantileEpsilon_) || (x > upperBound_ + quantileEpsilon_)) return 0.0;
   return normalizationFactor_ * distribution_.computePDF(point);
 }
 
 
 /* Get the CDF of the distribution: CDF_trunc = 1[a, b] * (CDF - CDF(a)) / (CDF(b) - CDF(a)) + 1]b, inf] */
 Scalar TruncatedDistribution::computeCDF(const Point & point) const
 {
-  if (point.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=1, here dimension=" << point.getDimension();
+  if (point.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=" << getDimension() << ", here dimension=" << point.getDimension();
+
+  for (UnsignedInteger k = 0; k < getDimension(); ++ k)
+  {
+    const Scalar xK = point[k];
+    if (xK <= bounds_.getLowerBound()[k]) return 0.0;
+    if (xK >= bounds_.getUpperBound()[k]) return 1.0;
```

Comment:
No. The CDF is zero if at least one of the components of the argument is less than the corresponding component of the lower bound, but is 1 only if ALL the components are greater or equal to the corresponding components of the upper bound. If only a part of the component is greater than the upper bound, then the CDF is obtained from the marginal distribution corresponding to the other components multiplied by the normalization factor. See NormalCopula::computeCDF to have an example of such a truncation.